src/hotspot/cpu/ppc/c1_Runtime1_ppc.cpp

```diff
@@ -603,10 +603,9 @@ OopMapSet* Runtime1::generate_code_for(C1StubId id, StubAssembler* sasm) {
       { // Support for uint StubRoutine::partial_subtype_check( Klass sub, Klass super );
         const Register sub_klass = R5,
                        super_klass = R4,
-                       temp1_reg = R6,
-                       temp2_reg = R0;
-        __ check_klass_subtype_slow_path(sub_klass, super_klass, temp1_reg, temp2_reg); // returns with CR0.eq if successful
-        __ crandc(CCR0, Assembler::equal, CCR0, Assembler::equal); // failed: CR0.ne
+                       temp1_reg = R6;
+        __ check_klass_subtype_slow_path(sub_klass, super_klass, temp1_reg, noreg); // may return with CR0.eq if successful
```

Comment:
The comment is unclear to me. Where is the result of the subtype check? Can it also return with CR0.ne if successful?
I noticed you added the `crandc` to `check_klass_subtype_slow_path_linear()` but if we reach there calling from this location then the `crandc` is not emitted because `L_success == nullptr`. Is this ok?
I'd appreciate comments on the masm methods explaining how the result of the subtype check is conveyed.

Reply:
The correct result is always in CR0 with this PR.
"return" means "blr", here. That can optionally be used in case of success. In this case, CR0 is always "eq".
I've moved the `crandc` instruction into `check_klass_subtype_slow_path_linear` which contains such a "blr" for a success case. This way, the linear version works exactly as before.
The new code `check_klass_subtype_slow_path_table` doesn't use "blr". That's why I added "may" to the comment.

Reply:
This is extremely hard to see.
L2154 with the "blr" in `check_klass_subtype_slow_path_linear` looks redundant to me. It should be removed if you agree.
The comment here should be adapted then too.
Also the comment at macroAssembler_ppc.cpp:2258 needs to be adapted because fallthrough from `check_klass_subtype_slow_path` does not mean "not successful". `L_failure` could be renamed to `L_fast_path_failure`

Reply:
I think `L_failure` is correct. And it's used the same way on all platforms.

Reply:
Ah, I see. I missed that `L_failure` is passed by reference
https://github.com/openjdk/jdk/blob/c00557f8f53ff729c8a1857a4ffcc585d3f8c6c4/src/hotspot/cpu/ppc/macroAssembler_ppc.cpp#L2159-L2168
Therefore it's never null and `L_failure` is reached if, and only if the result of the type check is negative.